zscan.go

```diff
@@ -962,3 +963,44 @@ func stringToNodeID(l lex) (uint64, *ParseError) {
 	}
 	return u, nil
 }
+
+// parse IPv4 address to four-byte slice under net.IP to save memory (see #166)
+func stringToIPv4(l lex) (net.IP, *ParseError) {
+	var (
+		ip  uint64
+		oct uint32
+		b   byte
+		num byte
+	)
+
+	// not checking len upfront, error will be obvious in soon anyway
+	for _, b = range []byte(l.token) {
```

Comment:
why does this need to be converted to []byte? We could use the string as-is, no? And convert a single value to a byte?